tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,184 @@
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manger --create-namespace --version v1.15.2 --set crds.enabled=true", timeout => 120);
+
+    sleep 180;
+    assert_script_run("helm list --all-namespaces");
+    assert_script_run("kubectl get pods --all-namespaces");
+
+    # Access to Application collection registery
+    # Get docker username and password
+    assert_script_run("kubectl create ns $namespace");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$SECRET_application_collection' -n $namespace", timeout => 120);
+
+    # Install private-ai-stack
+    my $gitlab_clone_url = 'https://git:' . $git_token . '@' . $repo_url;
+    assert_script_run("git clone $gitlab_clone_url");
+    assert_script_run("curl -o $vf_name $ai_chart_repo", timeout => 120);
+    assert_script_run("kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.28/deploy/local-path-storage.yaml", timeout => 120);
+    assert_script_run("helm upgrade --install suse-private-ai private-ai-charts --namespace $namespace --create-namespace --values $vf_name", timeout => 300);
+    assert_script_run("helm list --namespace $namespace");
+
+    # Check pod status and log for successful
+    my @out = split(' ', script_output("kubectl get pods --namespace $namespace -o custom-columns=':metadata.name'"));
+    record_info("Pod names", join(" ", @out));
+    foreach my $pod (@out) {
+        my $counter = 0;
+        while ($counter++ < 30) {
+            sleep 10;
+            my $status = script_output("kubectl get pod $pod -n $namespace -o=jsonpath='{.status.phase}'", proceed_on_failure => 1);
+            if ($status eq 'Running') {
+                record_info("$pod is running. ");
+                next;
+            }
+            my $logs = script_output("kubectl logs $pod -n $namespace", proceed_on_failure => 1);
+            if ($logs =~ /ERROR|FAILURE|Exception|Failed/) {
+                record_info("$pod failed due to error in log: $logs \n ");
+                next;
+            }
+        }
+    }
+    assert_script_run("kubectl get all --namespace $namespace");
+}
+
+sub test_openwebui_service {
+    my ($instance, $namespace) = @_;
+    my $sr_name = 'open-webui';
+    record_info('OpenWebUI service');
+
+    # Get open-webUI ipaddress and port and verify connectivity
+    my $ipaddr = script_output("kubectl get service $sr_name -n $namespace -o jsonpath='{.spec.clusterIP}'");
+    my $port = script_output("kubectl get service $sr_name -n $namespace -o jsonpath='{.spec.ports[0].port}'");
+    my $curl_cmd = "curl --max-time 10 --silent --output /dev/null https://$ipaddr:$port";
+    my $curl_result = script_run($curl_cmd);
```

Comment:
pls add comment on this command what's doing. 
Improve folow documenting steps.

Reply:
I will add more ..I wish you had reviewed most of the openqa codes which is missing documentation

Reply:
# After successfull installation, Get open-webUI ipaddress and port and verify connectivity